runtime/util_core/j9argscan.c

```diff
@@ -103,6 +105,31 @@ void scan_failed(J9PortLibrary * portLibrary, const char* module, const char *sc
 	j9tty_printf(PORTLIB, "<%s: unrecognized option --> '%s'>\n", module, scan_start);
 }
 
+/* Scan the next unsigned number off of the argument string.
+ * Store the result in *result
+ * Answer 0 on success
+ */
+uintptr_t scan_double(char **scan_start, double *result)
+{
+	char *endPtr = NULL;
+
+	*result = strtod(*scan_start, &endPtr);
+	if (ERANGE == errno) {
+		if (0.0 == *result) {
```

Comment:
You can not do it according spec: http://www.cplusplus.com/reference/cstdlib/strtod/
in case of ERANGE if return is not equal +/- HUGE_VAL this is underflow. The output value in this case is "value whose magnitude is no greater than the smallest normalized positive number" (not necessarily 0.0). If you want to handle it as zero *result should be set 0.0 before return with rc 0.

Reply:
As per man page of strtod:
*If the correct value would cause overflow, plus or minus HUGE_VAL (HUGE_VALF, HUGE_VALL) is returned (according to the sign of the value), and ERANGE is stored in errno. **If the correct value would cause underflow, zero is returned and ERANGE is stored in errno.** *
Based on that I consider it as underflow if `*return` is 0.0.

Reply:
From: http://pubs.opengroup.org/onlinepubs/007904975/functions/strtod.html
The Open Group Base Specifications Issue 6 IEEE Std 1003.1, 2004 Edition

"If the correct value would cause an underflow, a value whose magnitude is no greater than the smallest normalized positive number in the return type shall be returned and errno set to [ERANGE]."